src/App.tsx

```diff
@@ -21,12 +21,19 @@ function App() {
         <Routes>
           <Route path="/" element={<Homepage />} />
           <Route path="/coffeeshops" element={<CoffeeshopSearchPage />} />
-          <Route path="frequent-customer" element={<RestaurantList />}>
+          <Route
+            path="frequent-customer"
+            element={
+              <RestaurantProvider>
+                <RestaurantList />
+              </RestaurantProvider>
+            }
```

Comment:
wrapped the component that will be providing the id value in the ContextProvider 😤

Reply:
which now that you found that Outlet comes with Context we don't need. Brilliant work.